src/libraries/Microsoft.PowerFx.Core/Lexer/TexlLexer.cs

```diff
@@ -970,13 +980,35 @@ public LexerImpl(TexlLexer lex, string text, StringBuilder sb, Flags flags)
 
                 _numberIsFloat = flags.HasFlag(TexlLexer.Flags.NumberIsFloat);
                 _disableReservedKeywords = flags.HasFlag(TexlLexer.Flags.DisableReservedKeywords);
-
-                _modeStack = new Stack<LexerMode>();
-                _modeStack.Push(LexerMode.Normal);
+                _textFirst = flags.HasFlag(TexlLexer.Flags.TextFirst);
+
+                _modeStack = new Stack<LexerMode>();
+
+                if (_textFirst)
+                {
+                    int ich = 0;
+
+                    // skip whitespace looking for a leading '=' that starts a formula
+                    while (ich < _charCount && char.IsWhiteSpace(text[ich]))
+                    {
+                        ich++;
+                    }
+
+                    if (ich < _charCount && text[ich] == '=')
+                    {
+                        // skip '=' and treat the rest as a normal formula
+                        CurrentPos = ich + 1;
+                    }
+                    else
+                    {
+                        // CurrentPos doesn't move (stays at start of buffer) and treat all of it as text first
+                        _initialLexerMode = LexerMode.TextFirst;
+                    }
+                }
             }
 
             // If the mode stack is empty, this is already an parse, use NormalMode as a default
```

Comment:
>NormalMode [](http://example.com/codeflow?start=73&length=10)

update comment

Reply:
I don't understand this comment, can you say more?  I believe this logic is correct, in the case identified it should be TextFirst mode.

Reply:
The comment says that we use "Normal" mode as a default; after your changes we will use the initial lexer mode (normal or text first). The logic is correct, just the comment is a little stale

Reply:
Done.  Sorry, what a GitHub newbie I am.  I thought you were referring to the green changed code above.